### DISCUSSION TITLE
[Question]: How to get account's idToken from Auth0 to use with Hasura

### DISCUSSION THREAD

**Discussion by wilsonmsalberto** [nextauthjs/next-auth#2303] [Help]
### Question 💬

Hello! Currently trying to find a way to get the `idToken` that is inside `account` to use with the hook `useSession`.

Right now, my config looks like this:

```
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  site: process.env.SITE,
  providers: [
    Providers.Auth0({
      clientId: process.env.AUTH0_CLIENT_ID,
      clientSecret: process.env.AUTH0_CLIENT_SECRET,
      domain: process.env.AUTH0_DOMAIN,
      idToken: true
    })
  ],
  callbacks: {
    jwt: async (token) => {
      return Promise.resolve(token)
    },
    session: async (session, token) => {
      return Promise.resolve(session)
    }
  }
}

export default (req, res) => NextAuth(req, res, options)
```

Login works fine, accessing a user's details works fine, but I need to now access an Hasura API that is connected to Auth0. Both services are connected through Auth0 rules and they define the proper JWT. 

For that, I need to pass the `idToken` but I can't seem to figure out the config combination that actually allows me to pass it through to the `session` object.

If I copy the value that is inside `idToken` and use it to access my Hasura API, it works perfectly, just can't seem to find a way to pass it through.

Any pointers? Thanks!

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by cameronrdecker** [comment]
Hopefully you were able to find a solution by now, but this is how I wrote the session callback:

```
async session({session, token}) {
     session.accessToken = token.accessToken;
     session.idToken = token.idToken;
     return session;
                
 },
```
